### ISSUE TITLE
Update to support node-sass ^2.0.0 and Node 0.12

### ISSUE THREAD

**Issue by stevepaulo** [sass/node-sass-middleware#23]
I have forked the project and am trying to get tests to run after upgrading the package requirements to node-sass ^2.0.0, since this module will not install under Node 0.12.

Of course, I can't seem to get it to work or I would just submit a PR. Maybe you can figure out what's up?

**Comment by jerone** [comment]
Yeah, it would be great if this package supported Node 0.12 by updating node-sass.

**Comment by jerone** [comment]
I've got node-sass installed by forcing `node-sass@2.0.1`, but ran into errors after that:

```
C:\Users\XXX\node_modules\node-sass-middleware\node_modules\node-sass\lib\index.js:220
  options.data ? binding.render(options) : binding.renderFile(options);
                                                   ^
TypeError: Cannot convert undefined or null to object
    at Object.module.exports.render (C:\Users\XXX\node_modules\node-sass-middleware\node_modules\node-sass\lib\index.js:220:52)
    at C:\Users\XXX\node_modules\node-sass-middleware\middleware.js:114:17
    at fs.js:336:14
    at FSReqWrap.oncomplete (fs.js:99:15)
```

Still figuring out what's wrong.

**Comment by am11** [comment]
@jerone,  since we are currently focused on the bare-bone node-sass and might not be updating middleware frequently, however, I can give you some pointers which i hope will help you fixing this issue. Moreover, your PR is most  welcome! :zap:

In node-sass v2.0.0-beta, `renderFile` and `renderFileSync` functions were removed from the public API and the functionality was merged in corresponding async and sync variants:

* If you set `file` option, node-sass will resolve absolute path (if not already) and propagate it to LibSass which will read file contents and try compiling the Sass code.
* If you set `data` option, LibSass will consider the data value as Sass code and try to compile it.
* In case you set both, LibSass will give precedence to `data` and use `file` path for generating source-map (if sourceMap option is set).
  * If sourceMap option was set with `data` and input path (`file` option) was missing, LibSass will set input as `"stdin"` (string) in source-map's [`sources[]` member](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/).

That was considered as <kdb>breaking change</kdb> in v2.0.0-beta.

Also see:
* Node-Sass README: https://github.com/sass/node-sass/blob/master/README.md
* Issue where this architectural change was proposed and discussed: https://github.com/sass/node-sass/issues/547
* Release notes v2.0.0-beta: https://github.com/sass/node-sass/releases/tag/v2.0.0-beta
* Release notes v2.0.0 (stable): https://github.com/sass/node-sass/releases/tag/v2.0.0
* Related prospective changes: https://github.com/sass/node-sass/issues/676

**Comment by jerone** [comment]
@am11 Thanks for the info. I've got some code ready that makes `node-sass`middleware` working for me again. However I want to run the tests, but I haven't found out how to do it on Windows yet...

**Comment by am11** [comment]
This is fixed by 0918abd via #24. Credit @jerone! :tada:
